source/agent/addons/avstreamLib/AVStreamOutWrap.cc

```diff
@@ -134,10 +134,10 @@ void AVStreamOutWrap::New(const v8::FunctionCallbackInfo<v8::Value>& args)
             opts.dash_window_size = Nan::To<int32_t>(
                 Nan::Get(parameters, Nan::New("dashWindowSize").ToLocalChecked()).ToLocalChecked()).FromJust();
 
-            strncpy(opts.hls_method,
-                    getString(Nan::Get(parameters, Nan::New("method").ToLocalChecked()).ToLocalChecked()).c_str(),
-                    sizeof(opts.hls_method) - 1);
-            opts.dash_method[sizeof(opts.dash_method) - 1] = '\0';
+            const char *method_str = getString(Nan::Get(parameters, Nan::New("method").ToLocalChecked()).ToLocalChecked()).c_str() ;
+            uint32_t s = strlen(method_str) < sizeof(opts.dash_method) ?  strlen(method_str) : sizeof(opts.dash_method) -1 ;
+            memset(opts.dash_method, 0, sizeof(opts.dash_method));
+            strncat(opts.dash_method, method_str, s );
```

Comment:
Same as above one and other similar ones.
Could you use sizeof(opts.hls_method) - 1 instead of s here?